manifests/init.pp

```diff
@@ -36,6 +36,77 @@
   $vcl_conf             = '/etc/varnish/default.vcl',
 ) {
 
+  # validate parameters
+  validate_bool($manage_default_vcl)
+
+  validate_absolute_path($vcl_path)
+
+  if is_ip_address($admin_listen_address) == false {
+    fail("::varnish::admin_listen_address must be a valid IP address. Detected value is <${::varnish::admin_listen_address}>")
+  }
+
+  validate_re("${admin_listen_port}", '^\d+$', "::varnish::admin_list_port is not a valid server port. Detected value is <${::varnish::admin_list_port}>") # lint:ignore:only_variable_string
+  if $admin_listen_port == 0 {
+    fail("::varnish::admin_list_port is not a valid server port. Detected value is <${::varnish::admin_list_port}>")
+  }
+
+  if $listen_address != undef {
+    if is_ip_address($listen_address) == false {
+      fail("::varnish::listen_address must be a valid IP address or undef. Detected value is <${::varnish::listen_address}>")
+    }
+  }
+
+  validate_re("${listen_port}", '^\d+$', 'is not a valid server port') # lint:ignore:only_variable_string
+  if $listen_port == 0 {
+    fail("::varnish::listen_port is not a valid server port. Detected value is <${::varnish::listen_port}>")
+  }
+
+  validate_re("${min_threads}", '^\d+$', 'must be a non-zero integer') # lint:ignore:only_variable_string
+  if $min_threads == 0 {
+    fail("::varnish::min_threads must be a non-zero integer. Detected value is <${::varnish::min_threads}>")
+  }
+
+  validate_re("${max_threads}", '^\d+$', 'must be a non-zero integer') # lint:ignore:only_variable_string
+  if $max_threads == 0 {
+    fail("::varnish::max_threads must be a non-zero integer. Detected value is <${::varnish::max_threads}>")
+  }
+
+  validate_re("${thread_timeout}", '^\d+$', 'must be a non-zero integer') # lint:ignore:only_variable_string
+  if $thread_timeout == 0 {
+    fail("::varnish::thread_timeout must be a non-zero integer. Detected value is <${::varnish::thread_timeout}>")
+  }
+
+  validate_absolute_path($secret_file)
+
+  if is_string($storage) == false {
+    fail("::varnish::storage must be either file or malloc. Detected value is <${::varnish::storage}>")
+  }
+  validate_re("${storage}", '^[file|malloc]', "::varnish::storage must be either file or malloc. Detected value is <${::varnish::storage}>") # lint:ignore:only_variable_string
```

Comment:
storage is always a string, so you shouldnt have to type cast it with the double quotes and then ignore lint.